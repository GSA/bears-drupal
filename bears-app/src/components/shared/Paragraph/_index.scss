// ['regular', 'bold', 'extrabold', 'light', 'thin']
p.regular {
 font-weight: 400;
}

p.bold {
 font-weight: 700;
}

p.extrabold {
 font-weight: 800;
}

p.light {
 font-weight: 300;
}

p.thin {
 font-weight: 100;
}

