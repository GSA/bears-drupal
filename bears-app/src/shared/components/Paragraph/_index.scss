// ['regular', 'bold', 'extrabold', 'light', 'thin']
p,
p.regular {
  font-weight: 400;
  line-height: 162%;
}

p.bold {
  font-weight: 700;
}

p.extrabold {
  font-weight: 800;
}

p.light {
  font-weight: 300;
}

p.thin {
  font-weight: 100;
}
