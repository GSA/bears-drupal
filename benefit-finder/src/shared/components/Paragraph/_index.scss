// ['regular', 'bold', 'extrabold', 'light', 'thin']
@use '../../styles/mixins/_index.scss' as *;

#benefit-finder {
  p,
  p.regular {
    @include p;
  }

  p.bold {
    @include p;

    font-weight: 700;
  }

  p.extrabold {
    @include p;

    font-weight: 800;
  }

  p.light {
    @include p;

    font-weight: 300;
  }

  p.thin {
    @include p;

    font-weight: 100;
  }
}
