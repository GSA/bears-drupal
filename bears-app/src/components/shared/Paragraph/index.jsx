import { useHandleClassName } from '../hooks/useHandleClassName'
import PropTypes from 'prop-types'
import './_index.scss'

const Paragraph = ({ children, className, weight }) => {
  const defaultClasses = [`${weight}`]
  return (
    <p className={useHandleClassName({ className, defaultClasses })}>
      {children}
    </p>
  )
}

Paragraph.propTypes = {
  children: PropTypes.node || PropTypes.string,
  weight: PropTypes.oneOf(['regular', 'bold', 'extrabold', 'light', 'thin']),
}

export default Paragraph
