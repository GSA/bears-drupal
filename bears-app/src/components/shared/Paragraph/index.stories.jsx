import Paragraph from './index.jsx'

export default {
  component: Paragraph,
  tags: ['autodocs'],
  argTypes: {
    children: {
      control: 'text',
    },
    weight: {
      options: ['regular', 'bold', 'extrabold', 'light', 'thin'],
      control: { weight: 'radio' },
    },
  },
}

export const Regular = {
  args: {
    children:
      'Lorem ipsum dolor sit amet consectetur. Consectetur felis fermentum eu nunc elit imperdiet. Aenean porta sem sed nulla rutrum orci morbi non. Sollicitudin proin lorem magna ultrices gravida sem donec.',
    weight: 'regular',
  },
}

export const Bold = {
  args: {
    ...Regular.args,
    weight: 'bold',
  },
}

export const ExtraBold = {
  args: {
    ...Regular.args,
    weight: 'extrabold',
  },
}

export const Light = {
  args: {
    ...Regular.args,
    weight: 'light',
  },
}

export const Thin = {
  args: {
    ...Regular.args,
    weight: 'thin',
  },
}
