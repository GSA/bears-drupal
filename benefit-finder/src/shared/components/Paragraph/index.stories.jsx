import Paragraph from './index.jsx'

export default {
  component: Paragraph,
  tags: ['autodocs'],
  argTypes: {
    children: {
      control: 'text',
    },
    weight: {
      options: ['regular', 'bold', 'extrabold', 'light', 'thin'],
      control: { weight: 'radio' },
    },
  },
  args: {
    children:
      'Lorem ipsum dolor sit amet consectetur. Consectetur felis fermentum eu nunc elit imperdiet. Aenean porta sem sed nulla rutrum orci morbi non. Sollicitudin proin lorem magna ultrices gravida sem donec.',
  },
}

export const Regular = {
  args: {
    weight: 'regular',
  },
}

export const Bold = {
  args: {
    weight: 'bold',
  },
}

export const ExtraBold = {
  args: {
    weight: 'extrabold',
  },
}

export const Light = {
  args: {
    weight: 'light',
  },
}

export const Thin = {
  args: {
    weight: 'thin',
  },
}
