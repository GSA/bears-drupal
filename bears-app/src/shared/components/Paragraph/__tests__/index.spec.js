import renderer from 'react-test-renderer'
import Paragraph from '../index.jsx'

describe('Paragraph', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<Paragraph />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
