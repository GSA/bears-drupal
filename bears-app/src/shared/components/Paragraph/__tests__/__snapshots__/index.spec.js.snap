// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders a match to the previous snapshot 1`] = `
<p
  className=""
/>
`;
